src/_pytest/mark/expression.py

```diff
@@ -5,15 +5,18 @@
 expression: expr? EOF
 expr:       and_expr ('or' and_expr)*
 and_expr:   not_expr ('and' not_expr)*
-not_expr:   'not' not_expr | '(' expr ')' | ident ( '(' name '=' value ( ', ' name '=' value )*  ')')*
+not_expr:   'not' not_expr | '(' expr ')' | ident ('(' name '=' value ( ', ' name '=' value )*  ')')?
 
 ident:      (\w|:|\+|-|\.|\[|\]|\\|/)+
+name:       a valid ident, but not a reserved keyword
```

Comment:
i wonder if 
```suggestion
not_expr:   'not' not_expr | '(' expr ')' | ident kwargs?

ident:      (\w|:|\+|-|\.|\[|\]|\\|/)+
kwargs: ('(' name '=' value ( ', ' name '=' value )*  ')')
name:       a valid ident, but not a reserved keyword
```

helps with readability

Reply:
Done in 36b384afc7c2457a2772abcef1f3270a7067d0cb.

Reply:
@lovetheguitar *pro tip:* I recommend accepting suggested changes in PRs using GitHub UI and then pulling them to your computer. The reason is that this would mark the suggester as a co-author, and it's a good tone to credit things that collaborators proposed to their accounts.

Refs:
* https://x.com/hynek/status/1560196397885956097
*